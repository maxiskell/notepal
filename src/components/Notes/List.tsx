import React from "react";

import { INote } from "../../redux/types";
import Note from "./Note";

interface IListProps {
  notes: INote[];
}

const List: React.FC<IListProps> = ({ notes }) => (
  <div className="container" data-testid="note-list">
    {notes.map((note) => (
      <Note key={note.id} {...note} />
    ))}
  </div>
);

export default List;
