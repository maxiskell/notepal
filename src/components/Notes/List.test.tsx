import React from "react";
import { render } from "@testing-library/react";

import List from "./List";

const mockNotes = [
  {
    id: 1,
    title: "First Note",
    content: "Test Content",
  },
  {
    id: 2,
    title: "Another Note",
    content: "Some meaningless Content",
  },
];

test("render properly", () => {
  const { container } = render(<List notes={mockNotes} />);

  expect(container).toMatchSnapshot();
});

test("render all given notes", () => {
  const { getByText } = render(<List notes={mockNotes} />);

  expect(getByText(/First Note/)).toBeInTheDocument();
  expect(getByText(/Test Content/)).toBeInTheDocument();
  expect(getByText(/Another Note/)).toBeInTheDocument();
  expect(getByText(/Some meaningless Content/)).toBeInTheDocument();
});
