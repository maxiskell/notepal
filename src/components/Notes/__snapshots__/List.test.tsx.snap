// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render properly 1`] = `
<div>
  <div
    class="container"
    data-testid="note-list"
  >
    <div
      class="card w-100 mt-4"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          First Note
        </h5>
        <p
          class="card-text"
          style="white-space: pre-wrap;"
        >
          Test Content
        </p>
      </div>
    </div>
    <div
      class="card w-100 mt-4"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Another Note
        </h5>
        <p
          class="card-text"
          style="white-space: pre-wrap;"
        >
          Some meaningless Content
        </p>
      </div>
    </div>
  </div>
</div>
`;
